SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestMultiServerUnrealComponents/SpatialTestMultiServerUnrealComponents.cpp

```diff
@@ -0,0 +1,169 @@
+			AssertFalse(TestActor->HasAuthority(), TEXT("This server shouldn't have authority"));
+			if (AssertTrue(TestActor->AreAllDynamicComponentsValid(), TEXT("All dynamic components should have arrived")))
+			{
+				const bool bWrite = false;
+				const bool bOwnerOnlyExpected = true;
+				const bool bHandoverExpected = true;
+				const bool bInitialOnlyExpected = true;
+				ProcessActorProperties(bWrite, bOwnerOnlyExpected, bHandoverExpected, bInitialOnlyExpected);
+			}
+
+			FinishStep();
+		});
+
+	// Check on non-owning client that actor exists and properties are correct
+	AddStep(
+		TEXT("SpatialTestMultiServerUnrealComponents Validate Test Actor On Non-Owning Client"), FWorkerDefinition::Client(2),
+		[this]() -> bool {
+			return TestActor != nullptr;
+		},
+		[this]() {
+			AssertFalse(TestActor->HasAuthority(), TEXT("This client shouldn't have authority"));
+			if (AssertTrue(TestActor->AreAllDynamicComponentsValid(), TEXT("All dynamic components should have arrived")))
+			{
+				const bool bWrite = false;
+				const bool bOwnerOnlyExpected = false;
+				const bool bHandoverExpected = false;
+				const bool bInitialOnlyExpected = true;
+				ProcessActorProperties(bWrite, bOwnerOnlyExpected, bHandoverExpected, bInitialOnlyExpected);
+			}
+
+			FinishStep();
+		});
+}
+
+void ASpatialTestMultiServerUnrealComponents::ProcessActorProperties(bool bWrite, bool bOwnerOnlyExpected, bool bHandoverExpected,
+																	 bool bInitialOnlyExpected)
+{
+	auto Action = [&](int32& Source, int32 Expected) {
+		if (bWrite)
+		{
+			Source = Expected;
+		}
+		else
+		{
+			AssertEqual_Int(Source, Expected, TEXT("Property replicated incorrectly"));
+		}
+	};
+
+	Action(TestActor->StaticDataOnlyComponent->DataReplicatedVar, 10);
+
+	Action(TestActor->DynamicDataOnlyComponent->DataReplicatedVar, 20);
+
+	Action(TestActor->StaticDataAndHandoverComponent->DataReplicatedVar, 30);
+	Action(TestActor->StaticDataAndHandoverComponent->HandoverReplicatedVar, (!bWrite && !bHandoverExpected) ? 0 : 31);
+
+	Action(TestActor->DynamicDataAndHandoverComponent->DataReplicatedVar, 40);
+	Action(TestActor->DynamicDataAndHandoverComponent->HandoverReplicatedVar, (!bWrite && !bHandoverExpected) ? 0 : 41);
+
+	Action(TestActor->StaticDataAndOwnerOnlyComponent->DataReplicatedVar, 50);
+	Action(TestActor->StaticDataAndOwnerOnlyComponent->OwnerOnlyReplicatedVar, (!bWrite && !bOwnerOnlyExpected) ? 0 : 51);
+
+	Action(TestActor->DynamicDataAndOwnerOnlyComponent->DataReplicatedVar, 60);
+	Action(TestActor->DynamicDataAndOwnerOnlyComponent->OwnerOnlyReplicatedVar, (!bWrite && !bOwnerOnlyExpected) ? 0 : 61);
+
+	Action(TestActor->StaticDataAndInitialOnlyComponent->DataReplicatedVar, 70);
+	Action(TestActor->StaticDataAndInitialOnlyComponent->InitialOnlyReplicatedVar, (!bWrite && !bInitialOnlyExpected) ? 0 : 71);
+
+	Action(TestActor->DynamicDataAndInitialOnlyComponent->DataReplicatedVar, 80);
+	Action(TestActor->DynamicDataAndInitialOnlyComponent->InitialOnlyReplicatedVar,
```

Comment:
```suggestion
	// Initial only data is only expected if initial only is disabled, counterintuitively, because the
	// current implementation of initial only does not handle dynamic components,
	// but the SpatialGDK behavior without the bInitialOnlyEnabled flag is to send all data regardless.
	Action(TestActor->DynamicDataAndInitialOnlyComponent->InitialOnlyReplicatedVar,
```
As it took me a minute to parse the logic. Can be changed too.

Reply:
Yep fair - will add